engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/FlinkProcessRegistrar.scala

```diff
@@ -231,8 +235,15 @@ class FlinkProcessRegistrar(compileProcess: (EspProcess, ProcessVersion) => (Cla
 
         case part:SinkPart =>
           throw new IllegalArgumentException(s"Process can only use flink sinks, instead given: ${part.obj}")
-        case part@CustomNodePart(transformer:FlinkCustomStreamTransformation,
-              node, validationContext, nextParts, ends) =>
+        case CustomNodePart(transformerObj, node, validationContext, nextParts, ends) =>
+
+          val transformer = transformerObj match {
+            case t: FlinkCustomStreamTransformation => t
+            case ct: ContextTransformation =>
+              ct.implementation.asInstanceOf[FlinkCustomStreamTransformation]
```

Comment:
maybe we should fail more gracefully? if it's not FlinkCustomStreamTransformation?

Reply:
Good point, I'll throw Exception with some more meaningful message